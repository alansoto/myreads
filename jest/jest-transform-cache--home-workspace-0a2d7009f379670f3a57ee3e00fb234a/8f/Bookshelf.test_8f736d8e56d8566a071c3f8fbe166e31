/* istanbul ignore next */'use strict';var _jsxFileName = '/home/workspace/src/Bookshelf/Bookshelf.test.js';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme'); /* istanbul ignore next */var _enzyme2 = _interopRequireDefault(_enzyme);
var /* istanbul ignore next */_enzymeAdapterReact = require('enzyme-adapter-react-16'); /* istanbul ignore next */var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);
var /* istanbul ignore next */_Bookshelf = require('./Bookshelf'); /* istanbul ignore next */var _Bookshelf2 = _interopRequireDefault(_Bookshelf);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/* istanbul ignore next */_enzyme2.default.configure({ adapter: new /* istanbul ignore next */_enzymeAdapterReact2.default() });

describe('Bookshelf component', () => {
  it('renders without crashing', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Bookshelf2.default, /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: undefined }));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders bookshelfTitle from props', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Bookshelf2.default, /* istanbul ignore next */{ bookshelfTitle: 'My title', __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: undefined }));
    expect(wrapper.find('.bookshelf-title').text()).toBe('My title');
  });
});