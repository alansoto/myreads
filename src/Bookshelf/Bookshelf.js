import React, {Component} from 'react';
import Book from './../Book/Book';
import PropTypes from 'prop-types';

class Bookshelf extends Component {
  static propTypes = {
    bookshelfTitle: PropTypes.string
  }

  render() {
    const {bookshelfTitle} = this.props;

    return (
      <div>
        <div className="bookshelf">
          <h2 className="bookshelf-title">{bookshelfTitle}</h2>
          <div className="bookshelf-books">
            <ol className="books-grid">
              <Book/>
              <Book/>
              <Book/>

            </ol>
          </div>
        </div>
      </div>
    );
  }
}

export default Bookshelf;
