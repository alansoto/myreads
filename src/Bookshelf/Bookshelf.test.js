import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Bookshelf from './Bookshelf';

Enzyme.configure({adapter: new Adapter()});

const onUpdate = ()=>{console.log('hello')}
const testBookshelf = <Bookshelf bookshelfTitle='My title' onUpdate={onUpdate}/>

describe('Bookshelf component',()=>{
  it('renders without crashing', () => {
    const wrapper = shallow(testBookshelf);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders bookshelfTitle from props',()=>{
    const wrapper = shallow(testBookshelf);
    expect(wrapper.find('.bookshelf-title').text()).toBe('My title');
  });
});
