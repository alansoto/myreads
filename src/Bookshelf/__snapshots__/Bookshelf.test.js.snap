exports[`Bookshelf component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bookshelf
    bookshelfTitle="My title"
    onUpdate={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="bookshelf">
        <h2
                className="bookshelf-title">
                My title
        </h2>
        <div
                className="bookshelf-books">
                <ol
                        className="books-grid" />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="bookshelf-title">
            My title
</h2>,
          <div
            className="bookshelf-books">
            <ol
                        className="books-grid" />
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "My title",
            "className": "bookshelf-title",
          },
          "ref": null,
          "rendered": "My title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid" />,
            "className": "bookshelf-books",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "books-grid",
            },
            "ref": null,
            "rendered": null,
            "type": "ol",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bookshelf">
          <h2
                    className="bookshelf-title">
                    My title
          </h2>
          <div
                    className="bookshelf-books">
                    <ol
                              className="books-grid" />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="bookshelf-title">
              My title
</h2>,
            <div
              className="bookshelf-books">
              <ol
                            className="books-grid" />
</div>,
          ],
          "className": "bookshelf",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "My title",
              "className": "bookshelf-title",
            },
            "ref": null,
            "rendered": "My title",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <ol
                className="books-grid" />,
              "className": "bookshelf-books",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "books-grid",
              },
              "ref": null,
              "rendered": null,
              "type": "ol",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
